src/Fmt_ast.ml

```diff
@@ -34,6 +34,39 @@ let protect =
         first := false ) ;
       raise exc
 
+let scope c a =
+  let error {loc; txt} reason =
+    let w = Warnings.Attribute_payload (txt, reason) in
+    !Location.warning_printer loc Caml.Format.err_formatter w ;
+    c
+  in
+  match a with
+  | ({txt= "ocamlformat"} as aname), payload -> (
+    match payload with
+    | PStr
+        [ { pstr_desc=
+              Pstr_eval
+                ( { pexp_desc= Pexp_constant (Pconst_string (str, None))
+                  ; pexp_attributes= [] }
+                , [] ) } ] ->
+        List.fold ~init:c (String.split ~on:';' str) ~f:(fun c s ->
+            match String.lsplit2 ~on:'=' s with
+            | Some (name, value) -> (
+              try
+                Conf.update c ~name:(String.strip name)
+                  ~value:(String.strip value)
+              with e -> error aname (Exn.to_string e) )
+            | None ->
+                error aname "malformat option, should be [name=value]+" )
```

Comment:
s/malformat/malformed/ ?

Reply:
Would it make sense to share this parsing code with the code that processes .ocamlformat files?

Reply:
the syntax is currently different, are you suggesting to have the same syntax ?

Reply:
Yes, I think it would be easier for users if `--name=value` on the command line, `name=value` in .ocamlformat, and `name=value` as an attribute all meant the same and accepted as close to the same set of options as makes sense. The current config file format was a very quick hack and not really designed.

Reply:
I've refactored the code a bit.
The parsing logic is now shared between cmd-line, config file and attribute.
I also disallow some options in attributes (e.g margin)

Reply:
I've used the same format ocp-indent uses. That is, one line per option, key value separated by`=` (tolerate space for backward compatibility reason). Everything after a `#` is ignored